skxray/recip.py

```diff
@@ -194,3 +194,106 @@ def hkl_to_q(hkl_arr):
     """
 
     return np.linalg.norm(hkl_arr, axis=1)
+
+
+def q_rings(num_qs, first_q, delta_q, q_val, step_q=None):
+    """
+    This will find the indices of the required Q rings, find the bin
+    edges of the Q rings, and count the number of pixels in each Q ring.
+
+    Parameters
+    ----------
+    num_qs : int
+        number of Q rings
+
+    first_q : float
+        Q value of the first Q ring
+
+    delta_q : float
+        thickness of the Q ring
+
+    q_val : ndarray
+        Q space values for each pixel in the detector
+        shape is ([detector_size[0]*detector_size[1]], ) or
+        ([detector_size[0]*detector_size[1]], 1)
+
+    step_q : float, optional
+        step value for the next Q ring from the end of the previous Q ring
+
+    Returns
+    -------
+    q_inds : ndarray
+        indices of the Q values for the required rings
+
+    q_ring_val : ndarray
+        edge values of each Q ring
+        shape is (num_qs, 2)
+
+    num_pixels : ndarray
+        number of pixels in each Q ring
+    """
+
+    if (delta_q < 0):
+        raise ValueError("delta_q(thickness of the Q ring has to be positive")
+
+    q_val = np.asarray(q_val)
+
+    if (q_val.ndim == 1):
+        q_values = q_val
+    elif (q_val.ndim == 2):
+        q_values = np.ravel(q_val)
+    else:
+        raise ValueError("Q space values for each pixel in the detector"
+                         " has to be specified")
+
+    if (step_q is None):
+        # last Q ring edge value
+        last_q = first_q + num_qs*(delta_q)
+        # edges of all the Q rings
+        q_r = np.linspace(first_q, last_q, num=(num_qs+1))
+
+        # indices of Q rings
+        q_inds = np.digitize(q_values, np.array(q_r))
+        # discard the indices grater than number of Q rings
+        q_inds[q_inds > num_qs] = 0
+
+        # Edge values of each Q rings
+        q_ring_val = []
+
+        count = 0
```

Comment:
You don't need count here. Since you increment 1 in each loop, simply replace count with i.